src/ape/api/networks.py

```diff
@@ -1046,6 +1046,54 @@ def verify_chain_id(self, chain_id: int):
             raise NetworkMismatchError(chain_id, self)
 
 
+class ForkedNetworkAPI(NetworkAPI):
+    @property
+    def upstream_network(self) -> NetworkAPI:
+        """
+        The network being forked.
+        """
+        network_name = self.name.replace("-fork", "")
+        return self.ecosystem.get_network(network_name)
+
+    @property
+    def upstream_provider(self) -> Optional["UpstreamProvider"]:
+        """
+        The provider used when requesting data before the local fork.
+        Set this in your config under the network settings.
+        When not set, will attempt to use the default provider, if one
+        exists.
+        """
+
+        config_choice = self._network_config.get("upstream_provider")
+        if provider_name := config_choice or self.upstream_network.default_provider:
```

Comment:
shouldn't it always have a default?

Reply:
`.default_provider` is optional!

Let's say for example you are forking some network that does not have any providers associated with it. It would fail in that case.

I am not sure that is very easy in today's ecosystem... but with the plugin system, anything is possible.

Reply:
We could raise an error instead and change this to not be Optional? But technically, None fits the bill

Reply:
Yeah, I think the corner case that you have a network with no providers then it should just raise

Reply:
Ape does ship with networks without providers, so it is quite normal! But none of the upstream providers..
All the forked networks themselves are provider-less.

Reply:
done